src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -128,6 +129,56 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for .NET Framework.")]
+        public static void ZipArchiveEntry_BypassValidation()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                MemoryStream stream = populateStream().Result;
+                PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(s_tamperedFileName), stream, 8);  // patch uncompressed size in file header
+                using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+                 {
+                    // Set the AppContext Switch
+                    AppContext.SetSwitch("System.Compression.ZipArchiveEntry.SuppressValidation", true);
```

Comment:
Can we test it both ways? Once with the switch off and once with the switch on? When the switch is off, it should throw an exception. When it is turned on, it shouldn't.

Reply:
Good point